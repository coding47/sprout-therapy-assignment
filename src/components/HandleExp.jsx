import React, { useState } from 'react';
import calcOutput from '../utils/calcOutput';
import DisplayResult from './DisplayResult';

export default (props) => {
    const [A, setA ] = useState("false");
    const [B, setB ] = useState("false");
    const [C, setC ] = useState("false");
    const [D, setD ] = useState(0);
    const [E, setE ] = useState(0);
    const [F, setF ] = useState(0);
    const [result, setResult] = useState({});

    const handleSubmit = (event) => {
      console.log(props)
      event.preventDefault();
      setResult(calcOutput(A, B, C, D, E, F, props.SetRules));
    }
    
    return (
        <>
            <form onSubmit={handleSubmit}>
            <fieldset>
                <label htmlFor="A">A</label>
                <select id="A" value={A} onChange={(event) => setA(event.target.value)}>
                    <option value="true">true</option>
                    <option value="false">false</option>
                </select>
                <label htmlFor="B">B</label>
                <select id="B" value={B} onChange={(event) => setB(event.target.value)}>
                    <option value="true">true</option>
                    <option value="false">false</option>
                </select>
                <label htmlFor="C">C</label>
                <select id="C" value={C} onChange={(event) => setC(event.target.value)}>
                    <option value="true">true</option>
                    <option value="false">false</option>
                </select>
            </fieldset>
            <fieldset>
                <label htmlFor="D">D</label>
                <input id="D" type="number" step="0.1" value={D} onChange={(event) => setD(event.target.value)}/>
                <label htmlFor="E">E</label>
                <input id="E" type="number" value={E} onChange={(event) => setE(event.target.value)} />
                <label htmlFor="F">F</label>
                <input id="F" type="number" value={F} onChange={(event) => setF(event.target.value)}/>
            </fieldset>
            <input type="submit"/>
        </form>
        <DisplayResult Result={result} />
    </>
    )
}