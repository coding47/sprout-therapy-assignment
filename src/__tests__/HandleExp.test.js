import React from 'react';
import renderer from 'react-test-renderer';
import HandleExp from '../components/HandleExp';

it('renders correctly', () => {
  const tree = renderer
    .create(<HandleExp />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});