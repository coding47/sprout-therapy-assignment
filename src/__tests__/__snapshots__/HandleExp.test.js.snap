// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <fieldset>
      <label
        htmlFor="A"
      >
        A
      </label>
      <select
        id="A"
        onChange={[Function]}
        value="false"
      >
        <option
          value="true"
        >
          true
        </option>
        <option
          value="false"
        >
          false
        </option>
      </select>
      <label
        htmlFor="B"
      >
        B
      </label>
      <select
        id="B"
        onChange={[Function]}
        value="false"
      >
        <option
          value="true"
        >
          true
        </option>
        <option
          value="false"
        >
          false
        </option>
      </select>
      <label
        htmlFor="C"
      >
        C
      </label>
      <select
        id="C"
        onChange={[Function]}
        value="false"
      >
        <option
          value="true"
        >
          true
        </option>
        <option
          value="false"
        >
          false
        </option>
      </select>
    </fieldset>
    <fieldset>
      <label
        htmlFor="D"
      >
        D
      </label>
      <input
        id="D"
        onChange={[Function]}
        step="0.1"
        type="number"
        value={0}
      />
      <label
        htmlFor="E"
      >
        E
      </label>
      <input
        id="E"
        onChange={[Function]}
        type="number"
        value={0}
      />
      <label
        htmlFor="F"
      >
        F
      </label>
      <input
        id="F"
        onChange={[Function]}
        type="number"
        value={0}
      />
    </fieldset>
    <input
      type="submit"
    />
  </form>,
  <section>
    <h2>
      Result
    </h2>
    <p
      id="H"
    >
      H: 
    </p>
    <p
      id="K"
    >
      K: 
    </p>
  </section>,
]
`;
